[CI] MonitoringBulkRequestTests.testAddRequestContent

This reproduces, may have been caused by https://github.com/elastic/elasticsearch/pull/39888

On master 

https://elasticsearch-ci.elastic.co/job/elastic+elasticsearch+master+periodic/872/console

On 7.x

https://elasticsearch-ci.elastic.co/job/elastic+elasticsearch+7.x+matrix-java-periodic/ES_BUILD_JAVA=openjdk12,ES_RUNTIME_JAVA=java11,nodes=immutable&&linux&&docker/68/console

[build-periodic-7.x-68.txt.zip](https://github.com/elastic/elasticsearch/files/2958857/build-periodic-7.x-68.txt.zip)

```
  1> [2019-03-12T02:03:16,742][INFO ][o.e.x.m.a.MonitoringBulkRequestTests] [testAddRequestContent] before test                                                                                
  1> [2019-03-12T02:03:16,755][INFO ][o.e.x.m.a.MonitoringBulkRequestTests] [testAddRequestContent] after test                                                                                 
FAILURE 0.03s J7 | MonitoringBulkRequestTests.testAddRequestContent <<< FAILURES!              
   > Throwable #1: java.lang.AssertionError:   
   > Expected: "qxHU"                          
   >      but: was "_doc"                      
   >    at __randomizedtesting.SeedInfo.seed([3E7752DEB7B82078:6BF9E9DF11928659]:0)            
   >    at org.hamcrest.MatcherAssert.assertThat(MatcherAssert.java:20)                        
   >    at org.elasticsearch.xpack.monitoring.action.MonitoringBulkRequestTests.testAddRequestContent(MonitoringBulkRequestTests.java:136)                                                     
   >    at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)      
   >    at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)                                                                                    
   >    at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)                                                                            
   >    at java.base/java.lang.reflect.Method.invoke(Method.java:566)                          
   >    at java.base/java.lang.Thread.run(Thread.java:834)    
```

The seed is all that's needed, the compiler version doesn't matter

```
./gradlew :x-pack:plugin:monitoring:unitTest -Dtests.seed=3E7752DEB7B82078 -Dtests.class=org.elasticsearch.xpack.monitoring.action.MonitoringBulkRequestTests -Dtests.method="testAddRequestContent" -Dtests.security.manager=true -Dtests.locale=wo -Dtests.timezone=Chile/EasterIsland -Dcompiler.java=12 -Druntime.java=11
```

It looks like it's still trying to use custom types https://github.com/elastic/elasticsearch/blob/bec0a193675c648e162ce03c55b02305a5f2ba8d/x-pack/plugin/monitoring/src/test/java/org/elasticsearch/xpack/monitoring/action/MonitoringBulkRequestTests.java#L136

--- a/x-pack/plugin/monitoring/src/test/java/org/elasticsearch/xpack/monitoring/action/MonitoringBulkRequestTests.java
+++ b/x-pack/plugin/monitoring/src/test/java/org/elasticsearch/xpack/monitoring/action/MonitoringBulkRequestTests.java
@@ -76,7 +76,6 @@ public class MonitoringBulkRequestTests extends ESTestCase {
 
     public void testAddRequestContent() throws IOException {
         final XContentType xContentType = XContentType.JSON;
-        final String defaultType = rarely() ? randomAlphaOfLength(4) : null;
 
         final int nbDocs = randomIntBetween(1, 20);
         final String[] types = new String[nbDocs];
@@ -93,10 +92,10 @@ public class MonitoringBulkRequestTests extends ESTestCase {
                         if (rarely()) {
                             builder.field("_index", "");
                         }
-                        if (defaultType == null || randomBoolean()) {
-                            types[i] = randomAlphaOfLength(5);
-                            builder.field("_type", types[i]);
-                        }
+
+                        types[i] = randomAlphaOfLength(5);
+                        builder.field("_type", types[i]);
+
                         if (randomBoolean()) {
                             ids[i] = randomAlphaOfLength(10);
                             builder.field("_id", ids[i]);
@@ -133,7 +132,7 @@ public class MonitoringBulkRequestTests extends ESTestCase {
         int count = 0;
         for (final MonitoringBulkDoc bulkDoc : bulkDocs) {
             assertThat(bulkDoc.getSystem(), equalTo(system));
-            assertThat(bulkDoc.getType(), equalTo(types[count] != null ? types[count] : defaultType));
+            assertThat(bulkDoc.getType(), equalTo(types[count]));
             assertThat(bulkDoc.getId(), equalTo(ids[count]));
             assertThat(bulkDoc.getTimestamp(), equalTo(timestamp));
             assertThat(bulkDoc.getIntervalMillis(), equalTo(interval));

